### DISCUSSION TITLE
How to handle the automatically re-directing url

### DISCUSSION THREAD

**Discussion by sreenarayan** [cypress-io/cypress#28356] [Questions and Help]
Hi, In  my project they changed login page. Now baseUrl is automatically re-directing to other url. But, if I directly accessing the re-directing url this takes to different page not the original page. How to access this in cypress.  In cypress, it is not automatically re-directing. I have written the following code, but it is working for me.
Cypress.Commands.add('loginSession', (username, password) => {
  cy.session(
    [username, password],
    () => {
      // cy.visit('/app/zone');
      cy.origin("https:/abccd.com",{args : [username, password]},
      ([username],[password]) => {
      cy.visit("/idp/login_external/");
      cy.get(SIGNINPAGE.email).should("be.visible").type(username);
      cy.get(SIGNINPAGE.password).should("be.visible").type(password);
      SIGNINPAGE.signInButton();
      SELECTORS.searchAddress().should("be.visible",{timeout :60000});
    })
    },
  )
});
![image](https://github.com/cypress-io/cypress/assets/52318279/4de1c8bb-6085-4247-812d-7d4fbd638eab)

**Comment by adnanerlansyah403** [comment]
you can use the ones event handle on cypress like this :
```
cypress.on('url:changed', (url) => {
// the callback will get the newurl after the url was already changed or when moved into other page
const magicOrigin = new URL(newUrl) // we should change the url to instance URL to get the details of currentUrl like (host, hostname, path, etc.)
if(magicOrigin.host.includes('abccd.com')) {
 cy.origin(magicOrigin.url, () => {
   // do a process login
 })
 // continue the previous process on the first page or url.
}

})
```
